CMakeLists.txt

```diff
@@ -30,11 +30,11 @@ set(PLATFORM linux_x86)
 
 add_library(crossplatform_compiler_flags INTERFACE)
 target_compile_features(crossplatform_compiler_flags INTERFACE cxx_std_11)
-target_compile_options(crossplatform_compiler_flags INTERFACE -std=c++11 -Werror -Wall)
+target_compile_options(crossplatform_compiler_flags INTERFACE -std=c++11 -Werror -Wall -Wconversion)
```

Comment:
This is nice, thanks.

Once, here, could we add more? I found [this list from Zephyr project](https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/compiler/gcc/compiler_flags.cmake#L80-L89) very comprehensive.

Reply:
Oh man `-switch-default`! I even found that one several months back, but I forgot to submit it.

I can do it, but I'll follow up with a separate PR, because there will be many changes in this PR not related to the warnings too.

Reply:
Okey. We can also do it internally at some other point in time.

Reply:
Done here: https://github.com/mendersoftware/mender/pull/1677

Reply:
I was wrong about `switch-default` btw, and remember now why I didn't add it. See the PR.